drivers/spi/spi_nxp_lpspi/spi_nxp_lpspi.c

```diff
@@ -253,6 +252,11 @@ static void lpspi_isr(const struct device *dev)
 		lpspi_handle_tx_irq(dev);
 	}
 
+	if (spi_context_rx_len_left(ctx) == 0) {
+		base->IER &= ~LPSPI_IER_RDIE_MASK;
+		base->CR |= LPSPI_CR_RRF_MASK; /* flush rx fifo */
+	}
```

Comment:
is it worth it to maybe introduce another uint8_t or something in the `struct lpspi_driver_data` to have some status flags for the transfer, to avoid the extra unnecessary peripheral bus accesses during the isr? also unrelated to this PR but the VERID major version indicating the old stalling behavior probably should be stored similarly instead of checked every isr

Reply:
Probably. Will look into it tomorrow. The biggest performance killer is probably the watermarks of zero, though.

Reply:
@decsny Question: What happens if the RX fifo does not get flushed regularly? Would the entire SPI device deadlock once the RX fifo is full and nobody empties it?

Reply:
@decsny Yes, the answer is, the SPI device deadlocks if you don't clear the RX fifo regularly. Once it fills up, the transfer simply stops until you empty it. The fact that we disable the interrupts does not change anything; meaning, it was already being called repeatedly before this change.

Reply:
There is RXMSK bit but I don't want to mess with the masking bits, they have some unexpected effects

Reply:
As this is a bugfix PR, I'd not mix those two together. If needed, people can do optimization afterwards imo.